docs/admin/auth/oidc.md

```diff
@@ -0,0 +1,163 @@
+# OIDC Authentication
+
+With OIDC authentication, users can log in to your Coder deployment using your identify provider (e.g. KeyCloak, Okta, PingFederate, Azure AD, etc.)
+
+## Configuring
+
+### Step 1: Set Redirect URI with your OIDC provider
+
+Your OIDC provider will ask you for the following parameter:
+
+- **Redirect URI**: Set to `https://coder.domain.com/api/v2/users/oidc/callback`
+
+### Step 2: Configure Coder with the OpenID Connect credentials
+
+See [configuring Coder](../configure.md) to learn how to modify your server configuration depending on your platform (e.g. Kubernetes, system service, etc).
+
+Set the following environment variables for your Coder server:
+
+```console
+CODER_OIDC_ISSUER_URL="https://issuer.corp.com"
+CODER_OIDC_EMAIL_DOMAIN="your-domain-1,your-domain-2"
+CODER_OIDC_CLIENT_ID="533...des"
+CODER_OIDC_CLIENT_SECRET="G0CSP...7qSM"
+```
+
+> Refer to our section below if you wish to authenticate [with a PKI](#pki-authentication-optional) instead of a client secret.
+
+Restart your Coder server to apply this configuration. An OIDC authentication button should appear on your log in screen 🎉
+
+![Log in with OIDC button](https://user-images.githubusercontent.com/22407953/261882891-7aa2e922-5572-490f-992a-07126bad0161.png)
+
+## Provider Specific Notes
+
+Any OIDC provider should work with Coder. With that being said, we have some special notes for specific providers.
+
+- [Keycloak](./keycloak.md)
+- [Active Directory Federation Services (ADFS)](./adfs.md)
+
+## Group and Role Sync (enterprise)
+
+Learn how to do [group and role sync](group-role-sync.md) with COder.
+
+## How Coder Reads OIDC claims
+
+When a user logs in for the first time via OIDC, Coder will merge both
+the claims from the ID token and the claims obtained from hitting the
+upstream provider's `userinfo` endpoint, and use the resulting data
+as a basis for creating a new user or looking up an existing user.
+
+To troubleshoot claims, set `CODER_VERBOSE=true` and follow the logs
+while signing in via OIDC as a new user. Coder will log the claim fields
+returned by the upstream identity provider in a message containing the
+string `got oidc claims`, as well as the user info returned.
+
+The following information is also avalible via a [sequence diagram](https://raw.githubusercontent.com/coder/coder/138ee55abb3635cb2f3d12661f8caef2ca9d0961/docs/images/oidc-sequence-diagram.svg) for visual learners.
```

Comment:
Change to `main` branch now imo.

```suggestion
The following information is also avalible via a [sequence diagram](https://raw.githubusercontent.com/coder/coder/main/docs/images/oidc-sequence-diagram.svg) for visual learners.
```

Reply:
This is stored in docs/images. We plan to remove the absolute link and use relative link for images too.

Reply:
Sure